src/parser/flags.ts

```diff
@@ -35,13 +35,21 @@ export function boolean<T = boolean>(
   } as BooleanFlag<T>
 }
 
-export const integer = build({
-  parse: async input => {
-    if (!/^-?\d+$/.test(input))
-      throw new Error(`Expected an integer but received: ${input}`)
-    return Number.parseInt(input, 10)
-  },
-})
+export const integer = (opts: { min?: number; max?: number } & Partial<OptionFlag<number>> = {}): OptionFlag<number | undefined> => {
```

Comment:
It would be nice if min and max were also integers, maybe change type to `bigint`?